e3sm_to_cmip/__main__.py

```diff
@@ -651,25 +651,30 @@ def _run_info_mode(self):  # noqa: C901
         messages = []
 
         # if the user just asked for the handler info
-        if self.freq == "mon" and not self.input_path and not self.tables_path:
+        # TONY SAYS: Why does this mode even exist?  And why does it depend upon "freq = mon"?
+        if self.freq == "mon" and not self.input_path and not self.tables_path: # info mode 1
```

Comment:
```suggestion
        if self.freq == "mon" and not self.input_path and not self.tables_path:
```

Carried over from original, non-refactored code (with poor documentation).

https://github.com/E3SM-Project/e3sm_to_cmip/blob/e95a74bb03bee034dd7f6fd3050f3a354944a4bf/e3sm_to_cmip/util.py#L288-L301

I'm guessing this is the most basic version of "info" mode using the default `freq=mon"` that just prints out all of the matching defined handlers in `e3sm_to_cmip` based on the user-selected variable list. If the user passes a custom `freq` (e.g., "day" or "3hr"), this statement gets ignored.
https://github.com/E3SM-Project/e3sm_to_cmip/blob/e95a74bb03bee034dd7f6fd3050f3a354944a4bf/e3sm_to_cmip/util.py#L159-L164

Reply:
Also you can make comments related to PR reviews directly in the file diff under the [Files changed](https://github.com/E3SM-Project/e3sm_to_cmip/pull/252/files) tab, rather than in the code.

Reply:
@tomvothecoder You guess correctly (the most basic version of "info" mode using the default `freq = mon`).  The problem (I believe) is that if your variable-list contains (say) "pr", it returns multiple handlers, including "day".  This is an entire "mode" dedicated to the cause of saving the user the great burden of having to type "freq = mon".

`# noqa: C901`

Reply:
@tomvothecoder @chengzhuzhang 

In summary, we have 3 "info modes":

- (mode 1) exists largely to save the user typing `-f mon`, supplying cmor tables, and returns ALL of monthly handlers, but also occasional "day" handlers when a variable has multiple frequencies - and the user must be prepared to deal with it.
- (mode 2) that works except where ambiguous formulas for the SAME variable and frequency exist, requiring one apply mode 3.
- (mode 3) which involves supplying a native file so that the available variables can be examined. Still possible to have ambiguity if the native data contains multiple "solutions".

We need to eliminate ambiguity up-front.  Specifying "freq mon" should not additionally return results for "freq day", and the user should not need to know which modes will fail on which variables.  If "model-version" is a distinguishing criterion, it should be an attribute of the handlers.

Reply:
**SHEESH**.  By "committing the suggested change" - the  entire conversation was eliminated!

Reply:
@chengzhuzhang @tomvothecoder I have restored the ambiguous definitions for Amon rsut and rsutcs.  I will conducts extensive "info mode 3" testing to see if ambiguous results remains per native source data.  This means testing the "info mode" for different model versions and experiments, providing a source datafile for each.  Not hard to program, but will take several hours to run.  This process will need to be repeated when v3 data becomes available.

The script will be:

`/p/user_pub/e3sm/bartoletti1/Operations/9_Testing/e3sm_to_cmip/exhaustive_info_mode_3_testing.sh`

Reply:
Thank you for restoring the variables..